packages/firestore/src/local/indexeddb_query_cache.ts

```diff
@@ -97,36 +97,73 @@ export class IndexedDbQueryCache implements QueryCache {
     );
   }
 
-  addQueryData(
+  private saveQueryData(
     transaction: PersistenceTransaction,
     queryData: QueryData
   ): PersistencePromise<void> {
-    const targetId = queryData.targetId;
-    const addedQueryPromise = targetsStore(transaction).put(
-      this.serializer.toDbTarget(queryData)
-    );
-    if (targetId > this.metadata.highestTargetId) {
-      this.metadata.highestTargetId = targetId;
-      return addedQueryPromise.next(() =>
-        globalTargetStore(transaction).put(DbTargetGlobal.key, this.metadata)
-      );
-    } else {
-      return addedQueryPromise;
+    return targetsStore(transaction).put(this.serializer.toDbTarget(queryData));
+  }
+
+  private updateMetadataForQueryData(queryData: QueryData): boolean {
```

Comment:
I am not a big fan of this method name.

1) It seems very iOSy.
2) The name suggests that it also writes the metadata to IndexedDb.

Reply:
I left saving in a separate method since `removeQueryData` needs it as well, but does not need the same metadata update logic. 

I renamed `updateMetadataForQueryData` to `updateMetadata` and added a comment that it updates the in-memory copy and does not save.